### PR TITLE
[Bug 14521] Prevent a crash when clicking on menu items with submenus

### PR DESCRIPTION
**Pull request by livecodepanos**

### CODE REVIEW

File: engine/src/desktop-menu.cpp [hunk 1]
```diff
@@ -444,7 +444,8 @@ void MCButton::macopenmenu(void)
                 // SN-2014-08-25: [[ Bug 13240 ]] We need to keep the actual popup_menustring,
                 //  in case some menus are nested
                 MCStringRef t_menupick;
-                t_menupick = s_popup_menupick;
+                // PM-2015-02-09: [[ Bug 14521 ]] Make sure we don't pass nil to message_with_valueref_args
+                t_menupick = !MCStringIsEmpty(s_popup_menupick) ? s_popup_menupick : kMCEmptyString;
```
Comment (runrevmark) [inline engine/src/desktop-menu.cpp]: In what case is s_popup_menupick nil? Surely if the popup is not cancelled, MCPlatformPopUp returns true and therefore there must be something non-nil like in s_popup_menupick...

Comment (livecodepanos) [inline engine/src/desktop-menu.cpp]: Under the following scenario, MCPlatformPopUp returns true but s_popup_menupick is nil:
(Using develop-7.0 branch)
1. Create a new empty stack
2. Click "Inspector"
3. On the property inspector window that appears, click on the icon "Selection menu"
4. _Click_ on "Inspect" or "Property Profile"

Comment (runrevmark) [inline engine/src/desktop-menu.cpp]: I'm not clear on the recipe - do you mean the menu selection is cancelled?

If it is cancelled then I'm not sure it should be sending a menupick message at all...

**Comment by runrevmark** [comment]
Doesn't this affect 6.7 too?

**Comment by livecodepanos** [comment]
In 6.7 it does not crash. However, if you click (i.e if you select) on a menu item that has submenus, then the window will close, as if the popup was cancelled. So it might be better to submit this bugfix against 6.7, so as to have the same behaviour in both 6.7 and 7.0

**Comment by runrevmark** [comment]
Yes - we want the behavior to be the same as it was pre-6.7 in both 6.7 and 7.0 so if you could rebase this against 6.7 that would be great.

**Pull request closed without merging**